src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp

```diff
@@ -19,32 +19,257 @@
 #include "tools/common/XBUtilities.h"
 namespace XBU = XBUtilities;
 
+#include "core/common/device.h"
+#include "core/common/system.h"
+#include "core/common/error.h"
+#include "core/common/utils.h"
+#include "core/common/query_requests.h"
+
 // 3rd Party Library - Include Files
+#include <boost/format.hpp>
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
 
 // System - Include Files
 #include <iostream>
 
+
+// =============================================================================
+
+// ------ L O C A L   F U N C T I O N S ---------------------------------------
+
+namespace {
+
+enum configType {
+    CONFIG_SECURITY = 0,
+    CONFIG_CLK_SCALING,
+    CONFIG_CS_THRESHOLD_POWER_OVERRIDE,
+};
+
+enum memType {
+    UNKNOWN= 0,
+    DDR,
+    HBM, 
+    
```

Comment:
I suggest using enum classes and keep the enum enumerator names lower case.
E.g.  
`enum class config_type : uint16_t { config_security, config_clk_scaling, ...};`
The enum classes provide type safety and  the enumerator names are local to the enum.